packages/app/src/cli/services/webhook/trigger.ts

```diff
@@ -28,83 +25,36 @@ interface WebhookTriggerOptions {
+  const app: AppInterface = await loadApp({directory: flags.path, configName: flags.config})
   const developerPlatformClient: DeveloperPlatformClient =
-    flags.developerPlatformClient ?? selectDeveloperPlatformClient()
+    flags.developerPlatformClient ?? selectDeveloperPlatformClient({configuration: app.configuration})
+  const options: WebhookTriggerOptions = await validateAndCollectFlags(flags, developerPlatformClient, app)
 
-  // Validation and collection of flags
-  const validFlags = await validatedFlags(developerPlatformClient, flags)
-
-  // Request of prompts for missing flags
-  const options: WebhookTriggerOptions = await collectMissingFlags(developerPlatformClient, validFlags)
-
-  await sendSample(developerPlatformClient, options)
+  await sendSample(options)
 }
 
-async function validatedFlags(
-  developerPlatformClient: DeveloperPlatformClient,
+async function validateAndCollectFlags(
   flags: WebhookTriggerFlags,
-): Promise<WebhookTriggerFlags> {
-  const [deliveryMethod, address] = parseAddressAndMethod(flags)
-
-  const [apiVersion, topic] = await parseVersionAndTopic(developerPlatformClient, flags)
-
-  let clientSecret
-  if (isValueSet(flags.clientSecret)) {
-    // Flag overwrites any other secret
-    clientSecret = flags.clientSecret as string
-  }
-
-  const response: WebhookTriggerFlags = {
-    topic,
-    apiVersion,
-    deliveryMethod,
-    address,
-    clientSecret,
-  }
-
-  return response
-}
-
-async function collectMissingFlags(
   developerPlatformClient: DeveloperPlatformClient,
-  flags: WebhookTriggerFlags,
+  app: AppInterface,
 ): Promise<WebhookTriggerOptions> {
   const apiVersion = await collectApiVersion(developerPlatformClient, flags.apiVersion)
-
   const topic = await collectTopic(developerPlatformClient, apiVersion, flags.topic)
+  const [address, deliveryMethod] = await collectAddressAndMethod(flags.deliveryMethod, flags.address)
+  const clientCredentials = await collectCredentials(flags.clientId, flags.clientSecret, app, deliveryMethod)
 
-  const [deliveryMethod, address] = await collectAddressAndMethod(flags.deliveryMethod, flags.address)
-
-  const clientCredentials = await collectCredentials(developerPlatformClient, flags.clientSecret)
-
-  const options: WebhookTriggerOptions = {
+  return {
     topic,
     apiVersion,
     deliveryMethod,
     address,
-    clientSecret: clientCredentials.clientSecret as string,
+    apiKey: clientCredentials.clientId,
+    clientSecret: clientCredentials.clientSecret,
+    developerPlatformClient: clientCredentials.developerPlatformClient ?? developerPlatformClient,
   }
-
-  if (deliveryMethod === DELIVERY_METHOD.EVENTBRIDGE) {
-    if (isValueSet(clientCredentials.apiKey)) {
-      options.apiKey = clientCredentials.apiKey
-    } else {
-      options.apiKey = await collectApiKey(developerPlatformClient)
-    }
-  }
-
-  return options
-}
-
-export function isValueSet(value: string | undefined): boolean {
```

Comment:
This is equivalent to `if (value)`:

<img width="180" alt="17-52-vvlsi-9eiga" src="https://github.com/Shopify/cli/assets/14979109/54c44414-acbf-432f-ab2a-ce85c4bf2bcf">